src/TableFunctions/TableFunctionPostgreSQL.cpp

```diff
@@ -39,7 +39,10 @@ StoragePtr TableFunctionPostgreSQL::executeImpl(const ASTPtr & /*ast_function*/,
 ColumnsDescription TableFunctionPostgreSQL::getActualTableStructure(const Context & context) const
 {
     const bool use_nulls = context.getSettingsRef().external_table_functions_use_nulls;
-    auto columns = fetchPostgreSQLTableStructure(connection->conn(), remote_table_name, use_nulls);
+    auto columns = fetchPostgreSQLTableStructure(
+            connection->conn(),
+            remote_table_schema.empty() ? remote_table_name : remote_table_schema + '.' + remote_table_name,
```

Comment:
Do we need some escaping here?

Reply:
No, because it won't work then. I checked, it only works this way.

Reply:
So what will happen if `remote_table_schema` or `remote_table_name` contain dot?

Reply:
Yes, it works this way: `WHERE  attrelid = '"test_schema"."nice.table"'::regclass`.
I'll fix.